src/pages/content-edit/lib/delegator-validity.js

```diff
@@ -0,0 +1,147 @@
+import m from "mithril";
+import debounce from "lodash.debounce";
+
+import css from "../invalid-msg.css";
+
+function reqPrefix(name) {
+    return "Required: " + name;
+}
+
+export default function Validity(content) {
+    var v = this;
+        content = content;
+
+    v.handlersAttached = false;
+
+    // Private functions.
+    function show() {
+        var wasInvalid = content.get().ui.invalid;
+
+        content.toggleInvalid(true);
+        if(!wasInvalid) {
+            m.redraw(); // Only do once; avoid superfluous redraws.
+        }
+    }
+
+    function hide() {            
+        // CSS transition does the rest.
+        content.toggleInvalid(false);
+        m.redraw();
+    }
+
+    v.debounceFade = debounce(function() {
```

Comment:
Alright. I was just using the `controller` patterns out of habit, I guess, since that's how everything I was looking at was written.